feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -392,33 +392,32 @@ class BillingWrapper(
         withConnectedClient {
             log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
 
-            val queryActiveSubscriptionsResult = this.queryPurchases(SkuType.SUBS)
-            if (!queryActiveSubscriptionsResult.isSuccessful()) {
-                val billingResult = queryActiveSubscriptionsResult.billingResult
-                val purchasesError = billingResult.responseCode.billingResponseToPurchasesError(
-                    RestoreStrings.QUERYING_SUBS_ERROR.format(billingResult.toHumanReadableDescription())
-                )
-                onError(purchasesError)
-                return@withConnectedClient
-            }
+            this.queryPurchasesAsync(SkuType.SUBS) { activeSubsResult, activeSubsPurchases ->
+                if (!activeSubsResult.isSuccessful()) {
+                    val purchasesError = activeSubsResult.responseCode.billingResponseToPurchasesError(
+                        RestoreStrings.QUERYING_SUBS_ERROR.format(activeSubsResult.toHumanReadableDescription())
+                    )
+                    onError(purchasesError)
+                    return@queryPurchasesAsync
```

Comment:
This return is not needed anymore because there's nothing after the else. In any case, I would probably remove the else and return here. That way you can remove indentation in everything in the `else` block